### PR TITLE
Tooltip portal

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

This turned out to be easy now that tooltip is in context.

## Related Issue

https://github.com/recharts/recharts/pull/4435

https://github.com/recharts/recharts/issues/2458

## Motivation and Context

Make sure Tooltip always renders on top of other content, and Cursor always below

## How Has This Been Tested?

tests, storybooks

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: src/component/TooltipBoundingBox.tsx [hunk 1]
```diff
@@ -111,35 +111,18 @@ export class TooltipBoundingBox extends PureComponent<TooltipBoundingBoxProps, S
...
-     * The x and 0 and width and height are static and cover the whole SVG document, because
-     * the tooltip itself is positioned via context.
-     *
-     * pointer-events: none is necessary because this foreignObject covers the whole chart,
-     * and if it was catching mouse events it would steal them all before they reach the chart
-     * and tooltip would never open.
-     */
     return (
-      <g>
-        <foreignObject x="0" y="0" width="100%" height="100%" style={{ pointerEvents: 'none' }}>
```
Comment (PavelVanecek) [inline src/component/TooltipBoundingBox.tsx:129]: Thanks to portal, Tooltip is rendered in HTML again and the foreignObject is no longer required.

**Comment by ckifer** [comment]
image
Your description says the cursor is always below other content but it looks like it isn't from the storybook

File: src/component/Tooltip.tsx [hunk 2]
```diff
@@ -113,6 +124,13 @@ function TooltipInternal<TValue extends ValueType, TName extends NameType>(props
    */
   const finalIsActive = activeFromProps ?? activeFromContext;
   const [lastBoundingBox, updateBoundingBox] = useGetBoundingClientRect(undefined, [payload, finalIsActive]);
+
+  const tooltipPortal = portalFromProps ?? tooltipPortalFromContext;
+  const cursorPortal = useCursorPortal();
+  if (tooltipPortal == null || cursorPortal == null) {
```
Comment (ckifer) [inline src/component/Tooltip.tsx:130]: what if someone doesn't want a cursor?

**Review by ckifer — CHANGES_REQUESTED** [review]
Looks great! Just concerned about the ordering

File: src/component/Tooltip.tsx (hunk 2 above)
Comment (PavelVanecek) [reply src/component/Tooltip.tsx:130]: The portal will be there always, the cursor inside will return null

**Comment by PavelVanecek** [comment]
I think the screenshot is from build 863? I ran that one, then I realized I need to have a separate cursor portal, then I rerun. My bad, I should have generated a new diff, here it is: https://www.chromatic.com/test?appId=63da8268a0da9970db6992aa&id=66211568b767d61b79aa05d1

**Comment by PavelVanecek** [comment]
For comparison here's a codesandbox I made on 2.x version - the tooltip wasn't clipped then. https://codesandbox.io/p/sandbox/tall-tooltip-739pqy?file=%2Fsrc%2FApp.tsx%3A62%2C16

**Comment by ckifer** [comment]
Sweet sorry, I could've found out post-merge. Thanks!

**Review by ckifer — APPROVED** [review]

**Pull request merged**